miro_preprocessor/image_sorter/src/sorter_logic.py

```diff
@@ -6,23 +6,206 @@
 dictionary and returns an instance of ``Decision``.
 """
 
+import dateutil.parser
 import enum
+import re
+
+
+class Rules:
+    def __init__(self, collection, image_data):
+        self.collection = collection
+        self.image_data = image_data
+
+    _empty_title_strings = [
+        "-",
+        "--",
+        "Awaiting description",
+        "Awaiting removal from MIRO as duplicate",
+        "Awaiting captions and consent form",
+        "Awaiting catalogue details",
+        "Awaiting caption info",
+        "No info available about this object",
+        "No neg"
+    ]
+
+    _cc_accesses = [
+        "CC-0",
+        "CC-BY",
+        "CC-BY-NC",
+        "CC-BY-NC-ND",
+    ]
+
+    @staticmethod
+    def _normalise_string(s):
+        if s is not None:
+            s = s.lower().strip()
+        if s == "":
+            s = None
+        return s
+
+    def _get(self, key):
+        return self.image_data.get(key)
+
+    def _get_normalised(self, key):
+        return self._normalise_string(self._get(key))
+
+    def _compare(self, key, value):
+        return self._get_normalised(key) == self._normalise_string(value)
+
+    def _is_collection(self, collection_name):
+        return self._normalise_string(self.collection) == self._normalise_string(f"images-{collection_name}")
+
+    def _search(self, regex, key):
+        return re.search(regex, self._get_normalised(key))
+
+    def _is_blank(self, key):
+        return self._get_normalised(key) is None
+
+    def _key_matches(self, key, string_list):
+        return any([self._compare(key, string) for string in string_list])
+
+    def _parse_date(self, date_string):
+        return dateutil.parser.parse(date_string, dayfirst=True)
+
+    def is_collection(self, *args):
+        return any([self._is_collection(collection) for collection in args])
+
+    @property
+    def image_library_dept_is_Archives_and_Manuscripts(self):
+        return self._compare("image_library_dept", "Archives and Manuscripts")
+
+    @property
+    def image_library_dept_is_Public_programmes(self):
+        return self._compare("image_library_dept", "Public programmes")
+
+    @property
+    def image_tech_captured_mode_is_videodisc(self):
+        return self._compare("image_tech_captured_mode", "videodisc")
+
+    @property
+    def is_innopac_id_8_digits(self):
+        return self._search(r"[0-9]{7}[0-9|x]{1}", "image_innopac_id") is not None
```

Comment:
This regex is a bit too permissive: in particular, it matches `1234567|`, which surely isn't the intention? You don’t need the `|` in the second group: the following is sufficient:

```python
return self._search(r"[0-9]{7}[0-9x]{1}", "image_innopac_id")
```

That everything is in square brackets is enough to implicitly add the OR for you, so the `|` is treated as a literal.

Also, do we care about case?